enclave/crypto/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+if (UNIX OR USE_CLANGW)
+    set(PLATFORM_SRC
+        ../../common/sgx/rand.S)
+elseif (WIN32)
+    set(PLATFORM_SRC
+        ../../common/sgx/rand.asm)
+endif()
+
+add_library(oecrypto STATIC
+    ../../common/asn1.c
+    ../../common/cert.c
+    ../../common/kdf.c
+    asn1.c
+    cert.c
+    crl.c
+    ec.c
+    entropy.c
+    cmac.c
+    hmac.c
+    key.c
+    random.c
+    rsa.c
+    sha.c
+    ${PLATFORM_SRC})
+
+maybe_build_using_clangw(oecrypto)
+
+target_include_directories(oecrypto PRIVATE
+    ${PROJECT_SOURCE_DIR}/include/openenclave/corelibc)
```

Comment:
If this is the case, should this actually link against oelibc (or even just oelibc_includes)?

(I'd appreciate an update to the documented dependency graph to go along with this PR so we know what we're intending to change here.)

Reply:
oelibc isn't corelibc. corelibc is the minimal set of standard C functions used by oecore. oelibc is the musl C standard library that is built from oecore/oecrypto/etc.

Reply:
Gotcha, that makes sense. But let's still update the graph since we're adding and changing dependencies.

Reply:
Yup, ill update the graph.

Reply:
Thanks 🙏

Reply:
> oelibc isn't corelibc. corelibc is the minimal set of standard C functions used by oecore. oelibc is the musl C standard library that is built from oecore/oecrypto/etc. To use a kernel comparision, oelibc is the user-space libc, while corelibc is the libc used internally by the kernel.

@mikbras @CodeMonkeyLeet @paulcallen 
What Akash describes about corelibc in this issue was more or less my understanding as well: corelibc is the minimal set of **standard C functions** used by oecore. As per our discussion in #1732, this description is not exactly accurate. I would suggest that we consider a more appropriate name for corelibc since it is acting like a platform abstraction layer.

Reply:
@andschwa I updated the graph and cleaned it up a bit.